File: VB/GanttSample/DashboardData.vb
Imports System
Imports System.Collections.ObjectModel

Namespace Sample

    Public Class DashboardData

        Private _Data As ObservableCollection(Of Sample.DataRecord)

        Public Property Data As ObservableCollection(Of Sample.DataRecord)
            Get
                Return _Data
            End Get

            Private Set(ByVal value As ObservableCollection(Of Sample.DataRecord))
                _Data = value
            End Set
        End Property

        Public Sub New()
            Me.Data = Me.GetData()
        End Sub

        Private Function GetData() As ObservableCollection(Of Sample.DataRecord)
            Dim rnd As System.Random = New System.Random()
            Dim recordList As System.Collections.ObjectModel.ObservableCollection(Of Sample.DataRecord) = New System.Collections.ObjectModel.ObservableCollection(Of Sample.DataRecord)()
            For series As Integer = 1 To 4 - 1
                For argument As Integer = 1 To 5 - 1
                    Dim startDate As System.DateTime = System.DateTime.Now.AddDays(rnd.[Next](10))
                    Dim endDate As System.DateTime = startDate.AddDays(rnd.[Next](5))
                    recordList.Add(New Sample.DataRecord() With {.SubTask = "Sub Task " & argument, .GlobalTask = "Task " & series, .StartDate = startDate, .EndDate = endDate})
                Next
            Next

            Return recordList
        End Function
    End Class

    Public Class DataRecord

        Public Property GlobalTask As String

        Public Property SubTask As String

        Public Property StartDate As DateTime

        Public Property EndDate As DateTime
    End Class
End Namespace

File: CS/GanttSample/DashboardData.cs
using System;
using System.Collections.ObjectModel;

namespace Sample {
    public class DashboardData {
        public ObservableCollection<DataRecord> Data { get; private set; }

        public DashboardData() {
            Data = GetData();
        }
        ObservableCollection<DataRecord> GetData() {
            Random rnd = new Random();
            ObservableCollection<DataRecord> recordList = new ObservableCollection<DataRecord>();
            for(int series = 1; series < 4; series++) {
                for(int argument = 1; argument < 5; argument++) {
                    DateTime startDate = DateTime.Now.AddDays(rnd.Next(10));
                    DateTime endDate = startDate.AddDays(rnd.Next(5));
                    recordList.Add(new DataRecord() {
                        SubTask = "Sub Task " + argument,
                        GlobalTask = "Task " + series,
                        StartDate = startDate,
                        EndDate = endDate });
                }
            }
            return recordList;
        }
    }
    public class DataRecord {
        public string GlobalTask { get; set; }
        public string SubTask { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }
}
